### PR TITLE
fix: renamed storageclass to remove topolvm

### PR DESCRIPTION
**Pull request by nbalacha**
Replaced "topolvm-" with "odf-lvm" in the storageclass name.

Signed-off-by: N Balachandran <nibalach@redhat.com>

### CODE REVIEW

**Review by Yuggupta27 — COMMENTED** [review]
Do we need to update [here](https://github.com/red-hat-storage/lvm-operator/blob/1ed0e13d84f7f1b953c444f1655219878c19af5a/controllers/topolvm_storageclass.go#L16) too?

File: controllers/topolvm_storageclass.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ func getTopolvmStorageClasses(lvmCluster *lvmv1alpha1.LVMCluster) []*storagev1.S
 	for _, deviceClass := range lvmCluster.Spec.DeviceClasses {
 		storageClass := &storagev1.StorageClass{
 			ObjectMeta: metav1.ObjectMeta{
-				Name: fmt.Sprintf("topolvm-%s", deviceClass.Name),
+				Name: fmt.Sprintf("odf-lvm-%s", deviceClass.Name),
```
Comment (Madhu-1) [inline controllers/topolvm_storageclass.go:96]: @nbalacha not sure it a problem for upgrades update of storageclass name is not supported we need to delete and recreate it?

Comment (nbalacha) [reply controllers/topolvm_storageclass.go:96]: Upgrades are not in the picture as yet as this is the first downstream release.

**Comment by nbalacha** [comment]
> Do we need to update [here](https://github.com/red-hat-storage/lvm-operator/blob/1ed0e13d84f7f1b953c444f1655219878c19af5a/controllers/topolvm_storageclass.go#L16) too?

Not required. This is the name of the resource manager, not the storageclass.

File: controllers/topolvm_storageclass.go (hunk 1 above)
Comment (leelavg) [inline controllers/topolvm_storageclass.go:96]: - with this change, we are make the project more downstream-ish, as we already have code to determine k8s environment (openshift or other), how about we use that info to decide on `odf-lvm-` vs `topolvm-`?

Comment (nbalacha) [reply controllers/topolvm_storageclass.go:96]: I think a more generic sc name is probably fine for both upstream and downstream.

Comment (leelavg) [reply controllers/topolvm_storageclass.go:96]: ack, so we'll proceed with this name for now and later change it to generic then.

**Review by leelavg — APPROVED** [review]
/lgtm

**Pull request merged**